### ISSUE TITLE
Colorspace error?

### ISSUE THREAD

**Issue by MarjaE2** [jbarlow83/OCRmyPDF#231]
Ocrmypdf is giving the following error on some b/w files:
```
ERROR - Traceback (most recent call last):
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ruffus/task.py", line 751, in run_pooled_job_without_exceptions
    register_cleanup, touch_files_only)
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ruffus/task.py", line 567, in job_wrapper_io_files
    ret_val = user_defined_work_func(*params)
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pipeline.py", line 197, in repair_pdf
    pdfinfo = PdfInfo(output_file)
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 681, in __init__
    self._pages = _pdf_get_all_pageinfo(infile)
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 599, in _pdf_get_all_pageinfo
    return [PageInfo(pdf, n) for n in range(pdf.numPages)]
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 599, in <listcomp>
    return [PageInfo(pdf, n) for n in range(pdf.numPages)]
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 605, in __init__
    self._pageinfo = _pdf_get_pageinfo(pdf, pageno)
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 583, in _pdf_get_pageinfo
    shorthand=userunit_shorthand)]
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 581, in <listcomp>
    pageinfo['images'] = [im for im in
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 524, in _find_images
    yield from _find_regular_images(container, contentsinfo)
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 443, in _find_regular_images
    draw.shorthand)
  File "/usr/local/Cellar/ocrmypdf/5.6.2/libexec/lib/python3.6/site-packages/ocrmypdf/pdfinfo.py", line 310, in __init__
    if self._enc == Encoding.jpeg2000 else '?'
KeyError: <Colorspace.jpeg2000: 10>
```
Please include the command line and (if needed) a test file with your issue report.

`ocrmypdf -l ukr --force-ocr pdf.pdf pdfOCR.pdf`

If possible, please use a test file that we can include in future test cases (no personal information, no copyrighted material).

I get the error with the black and white pdf from here, not with the color pdf from the same page:

https://archive.org/details/vidrodzhennianat03vynnuoft

**Comment by jbarlow83** [comment]
Well, it's a bug... can't give high priority at the moment, unfortunately.

See here for some ideas on using Ghostscript to rewrite the PDF and replace all JPEG2000s with other compression. The issue is JPEG2000 and removing them will fix this. https://superuser.com/questions/171267/convert-pdf-file-with-images-in-jpeg2000-to-pdf-with-non-jpeg2000-images  - There are good reasons to avoid JPEG2000 anyway (complex format, heavily patented, poor standardization, not visually better than modern JPEG encoders).

Note to self: What that site calls black and white is in fact grayscale. This issue might be possibly specific to a grayscale JPEG2000 that lacks the (optional?) colorspace tag.

**Comment by stevepsmythe** [comment]
Ran ocrmypdf on a file with /JPXDecode which eventually produced a KeyError: <Colorspace.jpeg2000: 10> at line 309 of pdfinfo.py. Since replacing all JPEG2000 PDFs will take too long, I modified pdfinfo.py to work on my files.  
Changed:
`                self._color = FRIENDLY_COLORSPACE[Colorspace.jpeg2000] \
                    if self._enc == Encoding.jpeg2000 else '?'
` 
to:
`                self._color = FRIENDLY_COLORSPACE['/DeviceRGB'] \
                    if self._enc == Encoding.jpeg2000 else '?'
`   
and it runs without  raising a KeyError.
